internal/pkg/cli/env_delete.go

```diff
@@ -348,6 +372,80 @@ func (o *deleteEnvOpts) ensureRolesAreRetained() error {
 	return nil
 }
 
+// emptyBuckets returns nil if buckets were deleted successfully. Otherwise, returns the error.
+func (o *deleteEnvOpts) emptyBuckets() error {
+	envStack, err := o.rg.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: []*string{aws.String("cloudformation")},
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(deploy.EnvTagKey),
+				Values: []*string{aws.String(o.name)},
+			},
+			{
+				Key:    aws.String(deploy.AppTagKey),
+				Values: []*string{aws.String(o.appName)},
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("find env stack resource: %w", err)
+	}
+	if len(envStack.ResourceTagMappingList) > 1 {
+		return fmt.Errorf("ambiguous env stacks found")
+	} else if len(envStack.ResourceTagMappingList) < 1 {
+		return fmt.Errorf("no env stack found")
+	}
+	stackID := envStack.ResourceTagMappingList[0].ResourceARN
+
+	s3buckets, err := o.rg.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: []*string{aws.String("s3:bucket")},
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(envS3BucketStackNameTagKey),
+				Values: []*string{aws.String(stack.NameForEnv(o.appName, o.name))},
+			},
+			{
+				Key:    aws.String(envS3BucketStackIDTagKey),
+				Values: []*string{stackID},
+			},
+			{
+				Key:    aws.String(deploy.EnvTagKey),
+				Values: []*string{aws.String(o.name)},
+			},
+			{
+				Key:    aws.String(deploy.AppTagKey),
+				Values: []*string{aws.String(o.appName)},
+			},
+		},
+	})
+
+	if err != nil {
+		return fmt.Errorf("find s3 bucket resources: %w", err)
+	}
+
+	var failedBuckets []string
+	emptyBucketFailed := false
+	for _, resourceTagMapping := range s3buckets.ResourceTagMappingList {
+		bucketARN, err := arn.Parse(*resourceTagMapping.ResourceARN)
```

Comment:
`aws.StringValue` is preferred, because it returns `""` on `nil` pointer. The same `nil` pointer would have caused panic if dereferenced. 

I (personally) am perfectly fine with dereferencing a pointer  #directly if it can be logically inferred from the code that the pointer is guaranteed to be non-nil:
```go
if myPtr == nil {
  return
}
deref := *myPtr
```

This is not really the case though - you do have to look at the AWS doc to know that `ResourceARN` will be non-nil.

```suggestion
		bucketARN, err := arn.Parse(aws.StringValue(resourceTagMapping.ResourceARN))
```